worker/restore.go

```
@@ -92,8 +94,160 @@ func RunRestore(pdir, location, backupId string, key x.SensitiveByteSlice,
+		prefix := initKey.DataPrefix()
+		startKey := append(x.DataKey(attr, math.MaxUint64))
+
+		itOpt := badger.DefaultIteratorOptions
+		itOpt.AllVersions = true
+		itOpt.Reverse = true
+		itOpt.Prefix = prefix
+		itr := txn.NewIterator(itOpt)
+		defer itr.Close()
+		for itr.Seek(startKey); itr.Valid(); itr.Next() {
+			item := itr.Item()
+			// We expect only complete posting list.
+			x.AssertTrue(item.UserMeta() == posting.BitCompletePosting)
+			if err := fn(item); err != nil {
+				return err
+			}
+			break
+		}
+		return nil
+	})
+}
+
+func updateGqlSchema(cors [][]byte) error {
+	entry := &badger.Entry{}
+	var version uint64
+	err := getData("dgraph.graphql.schema", func(item *badger.Item) error {
+		var err error
+		entry.Key = item.KeyCopy(nil)
+		entry.Value, err = item.ValueCopy(nil)
+		if err != nil {
+			return err
+		}
+		entry.UserMeta = item.UserMeta()
+		entry.ExpiresAt = item.ExpiresAt()
+		version = item.Version()
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	if entry.Key == nil {
+		return nil
+	}
+
+	var corsBytes []byte
+	corsBytes = append(corsBytes, []byte("\n\n\n# Below schema elements will only work for dgraph"+
+		" versions >= 21.03. In older versions it will be ignored.")...)
+	for _, val := range cors {
+		corsBytes = append(corsBytes, []byte(fmt.Sprintf("\n# Dgraph.Allow-Origin \"%s\"",
+			string(val)))...)
+	}
+
+	// Append the cors at the end of GraphQL schema.
+	pl := pb.PostingList{}
+	if err = pl.Unmarshal(entry.Value); err != nil {
+		return err
+	}
+	pl.Postings[0].Value = append(pl.Postings[0].Value, corsBytes...)
+	entry.Value, err = pl.Marshal()
+	if err != nil {
+		return err
+	}
+
+	txn := pstore.NewTransactionAt(math.MaxUint64, true)
+	defer txn.Discard()
+	if err = txn.SetEntry(entry); err != nil {
+		return err
+	}
+	return txn.CommitAt(version, nil)
+}
+
+func getCors() ([][]byte, error) {
+	var corsVals [][]byte
+	err := getData("dgraph.cors", func(item *badger.Item) error {
+		val, _ := item.ValueCopy(nil)
```

Don't ignore the error.

---

Done.